tests/test_nn.py

```diff
@@ -778,6 +778,15 @@ def test_avgpool1d():
     assert jnp.all(output == answer)
 
 
+def test_adaptivepool1d():
+    x = jnp.arange(14).reshape(1, 14)
+    adaptive_pool = eqx.nn.AdaptiveAvgPool1D(4)
+    output = adaptive_pool(x)
+    answer = jnp.array([[1.5, 5.5, 9.0, 12.0]])
```

Comment:
**Inconsistency** 

For an input `(1, dim)`, other 1D methods return output of shape `(new_dim, )`. Adaptive1d returns `(1, new_dim)`.
What's the best output shape for this scenario?

Reply:
That's my mistake. I saw the test case: [link](https://github.com/patrick-kidger/equinox/blob/df5538f90c328100097a66eea1944051b2d684fc/tests/test_nn.py#L776) 
and the corresponding shape of the `answer`.